core/src/Streamly/Internal/Data/MutArray/Type.hs

```diff
@@ -3540,6 +3647,49 @@ strip eq arr@MutArray{..} = liftIO $ do
             else return cur
         else return cur
 
+-- | Strip elements which match the predicate, from the start of the array.
+--
+-- /Pre-release/
+{-# INLINE stripStart #-}
+stripStart :: forall a m. (Unbox a, MonadIO m) =>
+    (a -> Bool) -> MutArray a -> m (MutArray a)
+stripStart eq arr@MutArray{..} = liftIO $ do
+    st <- getStart arrStart
+    return arr {arrStart = st, arrEnd = arrEnd, arrBound = arrBound}
+
+    where
+
+    getStart cur = do
```

Comment:
getStart and getLast can possibly be moved out.
Perhaps `indexBy` and `indexByRev`.